lib/discovery.js

```diff
@@ -406,18 +408,12 @@ function mixinDiscovery(MySQL, mysql) {
 
   MySQL.prototype.buildPropertyType = function(columnDefinition) {
     var mysqlType = columnDefinition.dataType;
+    var columnType = columnDefinition.columnType;
     var dataLength = columnDefinition.dataLength;
 
     var type = mysqlType.toUpperCase();
     switch (type) {
       case 'CHAR':
-        if (dataLength === 1) {
-          // Treat char(1) as boolean
-          return 'Boolean';
-        } else {
-          return 'String';
-        }
-        break;
```

Comment:
First of I couldn't find any recommendation to use CHAR(1) as a data type for a boolean. Also I've found many situations where CHAR(1) is used to store a single letter (e.g. "m", "c", etc...) or as an old style bit array. Converting it to a boolean I believe is not the most common usecase.